### ISSUE TITLE
The "Attach Node Script" dialog should suggest snake_case file names, consistent with the documentation

### ISSUE THREAD

**Issue by homchom** [godotengine/godot-proposals#1211]
**Describe the project you are working on:** A 2D rhythm platformer

**Describe the problem or limitation you are having in your project:** The [GDScript Style Guide](https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_styleguide.html) page on the documentation says that per the style guide, GDScript file names should be in snake case. However, the 'Attach Node Script' dialog suggests the same name as the node, in pascal case by default.

![image](https://user-images.githubusercontent.com/41265646/87794171-41f99c00-c814-11ea-9dce-3aa996e4ee2c.png)

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:** To achieve consistency with the documentation, the autocompleted file name should be converted to snake case.

![image](https://user-images.githubusercontent.com/41265646/87794202-4cb43100-c814-11ea-83b3-afb5a6e799df.png)

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:** From what I have heard, Godot already has case conversion methods, so it should be easy to just convert pascal to snake case.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:** N/A

**Is there a reason why this should be core and not an add-on in the asset library?:** It is more consistent with the (core) documentation.

**Comment by Xrayez** [comment]
> From what I have heard, Godot already has case conversion methods, so it should be easy to just convert pascal to snake case.

Those methods exist, but they are not part of core currently, see my proposal #719.

**Comment by KoBeWi** [comment]
I think scene names should be snake_case too, otherwise it doesn't make much sense. And the script name already uses scene name, so they are consistent.

**Comment by Calinou** [comment]
To anyone implementing this: The convention should be changed automatically depending on the target language. It should *not* apply when using C#, which uses PascalCase for file names (the file name should be the same as the class name).

**Comment by Shadowblitz16** [comment]
pls no this should be a option not a requirement

**Comment by homchom** [comment]
> pls no this should be a option not a requirement

@Shadowblitz16 Both would still be an option. This is just suggesting that the default option should be consistent with the documentation.

**Comment by homchom** [comment]
> I think scene names should be snake_case too, otherwise it doesn't make much sense. And the script name already uses scene name, so they are consistent.

@KoBeWi I might be wrong, but doesn't it use the name of the node you attach it to? (which should be PascalCase)

**Comment by KoBeWi** [comment]
> I might be wrong, but doesn't it use the name of the node you attach it to? (which should be PascalCase)

Actually it's both. It uses scene name when script is attached to root and node name for other nodes.

**Comment by homchom** [comment]
That's funny. I just realized that and edited my comment, but before I saw your reply just now.

**Comment by MickL** [comment]
I wonder if the [docs](https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_styleguide.html#file-names) are wrong or the save dialog:

The docs say it should be snake_case but the dialog proposes PascalCase. Further, all tutorials I've seen so far, even the ones within the [Godot docs](https://docs.godotengine.org/en/stable/getting_started/step_by_step/your_first_game.html#finishing-up) use PascalCase.

**Comment by Calinou** [comment]
> I wonder if the docs are wrong or the save dialog:

We've established that the save dialog is wrong.

**Comment by passivestar** [comment]
I asked people and it seems like everybody is kinda doing their own thing when it comes to naming conventions:
https://twitter.com/passivestar_/status/1708115945799995542

There probably should be an additional dropdown on this screen for script names, with the default being whatever the documentation is recommending:

<img width="823" alt="image" src="https://github.com/godotengine/godot-proposals/assets/60579014/aa5ca3d6-1afc-4b5d-911e-e1d4fd772db9">

**Comment by GuyUnger** [comment]
i think many people (like me) started using pascal case for scripts because it’s what this box suggests. working with people and online resources that use different conventions is annoying, you can retype it if you really want to be different but i don’t think the engine should promote this with having a pascal case option

**Comment by ScriptSlinger** [comment]
After renaming 300 script file names, I support this as a potential feature, it would be nice to have a more consistent naming of script names according to the official convention stated by docs. It should be an option as well for people who want it to continue to default to PascalCase

**Comment by RedMser** [comment]
From the closed as duplicate #9199 there was a new suggestion which should be considered as well:

> A warning could be also added to the script creation windows to warn when a user's file name might cause issues dependent on the case specified in the project setting

Could be done as a follow-up PR, in which case this should likely be moved to a new dedicated design task.

**Comment by Dev-Marco** [comment]
The thing with this setting is that it only applies to scripts on nodes that are saved as a scene. So if save my BaseTower node as a scene and attach the script then it suggests the name `base_tower.gd`.
If the node is not saved as a scene it's `BaseTower.gd`.
I found that annoying so I created a plugin for anyone who's interested. It's my first Plugin so far and I am working on the comments :D
https://github.com/Dev-Marco/CCN---Correct-Conventional-Naming-for-Godot-Scripts.git
